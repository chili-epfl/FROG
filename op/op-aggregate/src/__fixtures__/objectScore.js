export default {
  _id: 'cjnyom0jn001b01rsv714f78l',
  activityData: {
    structure: 'individual',
    payload: {
      '7RwZeTAYvBsmwkuRR': {
        data: {
          cjnztobxc000001rscwxwedf1: {
            id: 'cjnztobxc000001rscwxwedf1',
            category: 'Yellow',
            score: 3,
            li: {
              liDocument: {
                liType: 'li-wordSelect',
                createdAt: '2018-11-02T09:34:31.777Z',
                createdBy: 'ac-select',
                payload: { word: 'ensure', color: '#FFFF00' }
              }
            }
          },
          cjnztobxd000j01rsau307wp2: {
            id: 'cjnztobxd000j01rsau307wp2',
            category: 'Yellow',
            score: 4,
            li: {
              liDocument: {
                liType: 'li-wordSelect',
                createdAt: '2018-11-02T09:34:31.777Z',
                createdBy: 'ac-select',
                payload: { word: 'ensure', color: '#FFFF00' }
              }
            }
          },
          cjnztobxd000101rsubxq49rp: {
            id: 'cjnztobxd000101rsubxq49rp',
            category: 'Yellow',
            score: 1,
            li: {
              liDocument: {
                liType: 'li-wordSelect',
                createdAt: '2018-11-02T09:34:31.777Z',
                createdBy: 'ac-select',
                payload: { word: 'cater', color: '#FFFF00' }
              }
            }
          },
          cjnztobxd000201rs6bi8xbkl: {
            id: 'cjnztobxd000201rs6bi8xbkl',
            category: 'Yellow',
            score: 5,
            li: {
              liDocument: {
                liType: 'li-wordSelect',
                createdAt: '2018-11-02T09:34:31.777Z',
                createdBy: 'ac-select',
                payload: { word: 'laid', color: '#FFFF00' }
              }
            }
          },
          cjnztobxd000301rsukqvu1ys: {
            id: 'cjnztobxd000301rsukqvu1ys',
            category: 'Yellow',
            score: 4,
            li: {
              liDocument: {
                liType: 'li-wordSelect',
                createdAt: '2018-11-02T09:34:31.777Z',
                createdBy: 'ac-select',
                payload: { word: 'food', color: '#FFFF00' }
              }
            }
          },
          cjnztobxd000401rs0su1c3l3: {
            id: 'cjnztobxd000401rs0su1c3l3',
            category: 'Yellow',
            score: 1,
            li: {
              liDocument: {
                liType: 'li-wordSelect',
                createdAt: '2018-11-02T09:34:31.777Z',
                createdBy: 'ac-select',
                payload: { word: 'ensure', color: '#FFFF00' }
              }
            }
          },
          cjnztobxd000501rsrmv4r6j4: {
            id: 'cjnztobxd000501rsrmv4r6j4',
            category: 'Yellow',
            score: -1,
            li: {
              liDocument: {
                liType: 'li-wordSelect',
                createdAt: '2018-11-02T09:34:31.777Z',
                createdBy: 'ac-select',
                payload: { word: 'ensure', color: '#FFFF00' }
              }
            }
          }
        }
      }
    }
  },
  socialStructure: {},
  globalStructure: {
    studentIds: [
      'bXaG7mBrMLhzpE9xi',
      'gnkSn7TXiTA9SYd8f',
      'KGFGwYrruhw3fMbwv',
      'C3NCZvKZFg3hnhqvs',
      'XcSgaxhb3yuDNTNZq',
      'YQjgzcgQGM2ELxJwE',
      'qkeubQS58zZJDMMdZ',
      'iWarW8gPJR3sis3vT',
      'yNjrjBftaxbJqb5Kv',
      'yPKouhEwhoQxd8N7L',
      'MMtir45Jz9SfRDnMh',
      'pa9KWNjffgAqQTX3N',
      'cF2qxcPSYLGWfwZ8e',
      'RS5zQwax4MXnjcwhA',
      'ZbCnakkuhEGjzYENs',
      'eWWzXBLW8tbo4qFp3',
      'pPJgig96WpR3yaLb6',
      '7RwZeTAYvBsmwkuRR',
      'pT7g88TndMGFuYncT'
    ],
    students: {}
  }
};
