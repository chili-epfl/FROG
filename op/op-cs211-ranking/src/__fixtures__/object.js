export const object = {
  activityData: {
    cjgdgp46h001kblj4cwsqtmrk: {
      structure: 'individual',
      payload: {
        notfirst: {
          config: {},
          data: {
            justification: 'sfdsdfsdfsdffdfff',
            answers: {
              notfirst: {}
            },
            group: {
              notfirst: 'Never Again'
            },
            msg:
              'Rina ranked the interfaces in the following order: train, drag, wave, jump, with the justification "sfdsdfsdfsdffdfff".'
          }
        },
        neveragain: {
          config: {},
          data: {
            justification: 'sfdsdfsdfsdffdfff',
            answers: {
              neveragain: {
                train: 1,
                drag: 2,
                wave: 3,
                jump: 4
              }
            },
            group: {
              neveragain: 'Never Again'
            },
            msg:
              'Rina ranked the interfaces in the following order: train, drag, wave, jump, with the justification "sfdsdfsdfsdffdfff".'
          }
        },
        AhRFh8ZDGroQ5DzdK: {
          config: {},
          data: {
            justification: 'sfdsdfsdfsdffdfff',
            answers: {
              AhRFh8ZDGroQ5DzdK: {
                train: 1,
                drag: 2,
                wave: 3,
                jump: 4
              }
            },
            group: {
              AhRFh8ZDGroQ5DzdK: 'Rina'
            },
            msg:
              'Rina ranked the interfaces in the following order: train, drag, wave, jump, with the justification "sfdsdfsdfsdffdfff".'
          }
        },
        EWwaMdYKGDZym44Tt: {
          config: {},
          data: {
            justification: 'ff',
            answers: {
              EWwaMdYKGDZym44Tt: {
                wave: 2,
                drag: 3,
                train: 1,
                jump: 4
              }
            },
            group: {
              EWwaMdYKGDZym44Tt: 'Jan'
            },
            msg:
              'Jan ranked the interfaces in the following order: wave, drag, train, jump, with the justification "ff".'
          }
        },
        xArWxtGksryMJBjh8: {
          config: {},
          data: {
            justification: 'ff',
            answers: {
              xArWxtGksryMJBjh8: {
                drag: 1,
                wave: 2,
                jump: 3,
                train: 4
              }
            },
            group: {
              xArWxtGksryMJBjh8: 'Trine'
            },
            msg:
              'Trine ranked the interfaces in the following order: drag, wave, jump, train, with the justification "ff".'
          }
        },
        xpAkdQ72YYsbthfbf: {
          config: {},
          data: {
            justification: 'sdfsdf',
            answers: {
              xpAkdQ72YYsbthfbf: {
                jump: 1,
                wave: 2,
                drag: 3,
                train: 4
              }
            },
            group: {
              xpAkdQ72YYsbthfbf: 'Ole'
            },
            msg:
              'Ole ranked the interfaces in the following order: jump, wave, drag, train, with the justification "sdfsdf".'
          }
        }
      }
    },
    cjgdgp46i001mblj4u5lfdd4r: {
      structure: {
        groupingKey: 'group'
      },
      payload: {
        '1': {
          config: {},
          data: {
            completed: true,
            justification: 'so far',
            answers: {
              xArWxtGksryMJBjh8: {
                jump: 1,
                wave: 2,
                drag: 3,
                train: 4
              },
              EWwaMdYKGDZym44Tt: {
                jump: 1,
                wave: 2,
                drag: 3,
                train: 4
              }
            },
            group: {
              xArWxtGksryMJBjh8: 'Trine',
              EWwaMdYKGDZym44Tt: 'Jan'
            },
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "so far".'
          }
        },
        '2': {
          config: {},
          data: {
            completed: true,
            justification: 'yeah',
            answers: {
              AhRFh8ZDGroQ5DzdK: {
                train: 1,
                drag: 2,
                wave: 3,
                jump: 4
              },
              xpAkdQ72YYsbthfbf: {
                train: 1,
                drag: 2,
                wave: 3,
                jump: 4
              }
            },
            group: {
              AhRFh8ZDGroQ5DzdK: 'Rina',
              xpAkdQ72YYsbthfbf: 'Ole'
            },
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "yeah".'
          }
        }
      }
    },
    cjgdgp46h001lblj4whyzno4p: {
      structure: {
        groupingKey: 'group'
      },
      payload: {
        '1': {
          config: {},
          data: {
            justification: 'done',
            answers: {
              xArWxtGksryMJBjh8: {
                wave: 1,
                jump: 2,
                drag: 3,
                train: 4
              },
              EWwaMdYKGDZym44Tt: {
                wave: 1,
                jump: 2,
                drag: 4,
                train: 3
              }
            },
            group: {
              xArWxtGksryMJBjh8: 'Trine',
              EWwaMdYKGDZym44Tt: 'Jan'
            },
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "done".'
          }
        },
        '2': {
          config: {},
          data: {
            justification: 'final',
            completed: true,
            answers: {
              xpAkdQ72YYsbthfbf: {
                drag: 1,
                train: 2,
                wave: 3,
                jump: 4
              },
              AhRFh8ZDGroQ5DzdK: {
                drag: 1,
                train: 2,
                wave: 3,
                jump: 4
              }
            },
            group: {
              xpAkdQ72YYsbthfbf: 'Ole',
              AhRFh8ZDGroQ5DzdK: 'Rina'
            },
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "final".'
          }
        }
      }
    }
  },
  socialStructure: {},
  globalStructure: {
    studentIds: [
      'EWwaMdYKGDZym44Tt',
      'xArWxtGksryMJBjh8',
      'xpAkdQ72YYsbthfbf',
      'AhRFh8ZDGroQ5DzdK'
    ],
    students: {
      EWwaMdYKGDZym44Tt: 'Jan',
      xArWxtGksryMJBjh8: 'Trine',
      xpAkdQ72YYsbthfbf: 'Ole',
      AhRFh8ZDGroQ5DzdK: 'Rina'
    }
  }
};
