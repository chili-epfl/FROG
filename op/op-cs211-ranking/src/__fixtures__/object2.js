export const object2 = {
  _id: 'cjgdu4sti00071sj4le1ry8kx',
  activityData: {
    cjgdu4sti00021sj4sy5xdedy: {
      structure: 'individual',
      payload: {
        '2tKssy45Qerurvz5T': {
          data: {
            justification: 'd',
            answers: {
              '2tKssy45Qerurvz5T': {
                plane: 1,
                automotive: 2,
                swimming: 3,
                train: 4
              }
            },
            group: {
              '2tKssy45Qerurvz5T': 'Chen Xi'
            },
            alert: false,
            completed: true,
            msg:
              'Chen Xi ranked the interfaces in the following order: plane, automotive, swimming, train, with the justification "d".'
          }
        },
        '6ZLem6dL9CkgLpLBE': {
          data: {
            justification: 'd',
            answers: {
              '6ZLem6dL9CkgLpLBE': {
                plane: 1,
                automotive: 2,
                swimming: 3,
                train: 4
              }
            },
            group: {
              '6ZLem6dL9CkgLpLBE': 'Ingunn'
            },
            msg:
              'Ingunn ranked the interfaces in the following order: plane, automotive, swimming, train, with the justification "d".'
          }
        },
        DAeNWt6uf5y3Kjwqx: {
          data: {
            justification: '',
            answers: {},
            group: {},
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "".'
          }
        },
        DZcvDkx5j3FuG9Zk9: {
          data: {
            justification: 'e',
            answers: {
              DZcvDkx5j3FuG9Zk9: {
                swimming: 1,
                automotive: 2,
                plane: 3,
                train: 4
              }
            },
            group: {
              DZcvDkx5j3FuG9Zk9: 'Kris'
            },
            alert: false,
            completed: true,
            msg:
              'Kris ranked the interfaces in the following order: swimming, automotive, plane, train, with the justification "e".'
          }
        },
        Fff6jCHLLk3aB9oij: {
          data: {
            justification: 'f',
            answers: {
              Fff6jCHLLk3aB9oij: {
                plane: 1,
                automotive: 2,
                train: 3,
                swimming: 4
              }
            },
            group: {
              Fff6jCHLLk3aB9oij: 'Are'
            },
            alert: false,
            completed: true,
            msg:
              'Are ranked the interfaces in the following order: plane, automotive, train, swimming, with the justification "f".'
          }
        },
        kLDRxiAyQeo5k2rNu: {
          data: {
            justification: 'f',
            answers: {
              kLDRxiAyQeo5k2rNu: {
                train: 1,
                plane: 2,
                automotive: 3,
                swimming: 4
              }
            },
            group: {
              kLDRxiAyQeo5k2rNu: 'Pierre'
            },
            alert: false,
            completed: true,
            msg:
              'Pierre ranked the interfaces in the following order: train, plane, automotive, swimming, with the justification "f".'
          }
        },
        nDG9FhEMBsJNt9Mwf: {
          data: {
            justification: 'c',
            answers: {
              nDG9FhEMBsJNt9Mwf: {
                swimming: 1,
                plane: 2,
                automotive: 3,
                train: 4
              }
            },
            group: {
              nDG9FhEMBsJNt9Mwf: 'Trine'
            },
            alert: false,
            completed: true,
            msg:
              'Trine ranked the interfaces in the following order: swimming, plane, automotive, train, with the justification "c".'
          }
        },
        nWnw9pm9mnrkKNGm3: {
          data: {
            justification: '',
            answers: {},
            group: {},
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "".'
          }
        }
      }
    },
    cjgdu4sti00041sj4v5qch1gk: {
      structure: {
        groupingKey: 'group'
      },
      payload: {
        '1': {
          data: {
            justification: '',
            answers: {
              '2tKssy45Qerurvz5T': {}
            },
            group: {
              '2tKssy45Qerurvz5T': 'Chen Xi'
            },
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "".'
          }
        },
        '2': {
          data: {
            justification: '',
            answers: {
              DZcvDkx5j3FuG9Zk9: {
                train: 1
              }
            },
            group: {
              DZcvDkx5j3FuG9Zk9: 'Kris'
            },
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "".'
          }
        },
        '3': {
          data: {
            justification: 'yes',
            answers: {
              '6ZLem6dL9CkgLpLBE': {
                train: 1,
                plane: 2,
                automotive: 3,
                swimming: 4
              },
              Fff6jCHLLk3aB9oij: {
                train: 1,
                plane: 2,
                automotive: 3,
                swimming: 4
              }
            },
            group: {
              '6ZLem6dL9CkgLpLBE': 'Ingunn',
              Fff6jCHLLk3aB9oij: 'Are'
            },
            alert: false,
            completed: true,
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "yes".'
          }
        },
        '4': {
          data: {
            justification: 'no',
            answers: {
              kLDRxiAyQeo5k2rNu: {
                swimming: 1,
                automotive: 2,
                plane: 3,
                train: 4
              },
              nDG9FhEMBsJNt9Mwf: {
                swimming: 1,
                automotive: 2,
                plane: 3,
                train: 4
              }
            },
            group: {
              kLDRxiAyQeo5k2rNu: 'Pierre',
              nDG9FhEMBsJNt9Mwf: 'Trine'
            },
            alert: false,
            completed: true,
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "no".'
          }
        }
      }
    },
    cjgdu4sti00031sj47sjionh9: {
      structure: {
        groupingKey: 'group'
      },
      payload: {
        '1': {
          data: {
            justification: '',
            answers: {
              '2tKssy45Qerurvz5T': {}
            },
            group: {
              '2tKssy45Qerurvz5T': 'Chen Xi'
            },
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "".'
          }
        },
        '2': {
          data: {
            justification: '',
            answers: {
              DZcvDkx5j3FuG9Zk9: {
                plane: 1
              }
            },
            group: {
              DZcvDkx5j3FuG9Zk9: 'Kris'
            },
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "".'
          }
        },
        '3': {
          data: {
            justification: 'ff',
            answers: {
              Fff6jCHLLk3aB9oij: {
                train: 1,
                plane: 2,
                automotive: 3,
                swimming: 4
              },
              '6ZLem6dL9CkgLpLBE': {
                train: 1,
                plane: 2,
                automotive: 3,
                swimming: 4
              }
            },
            group: {
              Fff6jCHLLk3aB9oij: 'Are',
              '6ZLem6dL9CkgLpLBE': 'Ingunn'
            },
            alert: false,
            completed: true,
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "ff".'
          }
        },
        '4': {
          data: {
            justification: 'f',
            answers: {
              nDG9FhEMBsJNt9Mwf: {
                swimming: 1,
                automotive: 2,
                plane: 3,
                train: 4
              },
              kLDRxiAyQeo5k2rNu: {
                swimming: 1,
                automotive: 2,
                plane: 3,
                train: 4
              }
            },
            group: {
              nDG9FhEMBsJNt9Mwf: 'Trine',
              kLDRxiAyQeo5k2rNu: 'Pierre'
            },
            alert: false,
            completed: true,
            msg:
              'undefined ranked the interfaces in the following order: , with the justification "f".'
          }
        }
      }
    }
  },
  socialStructure: {},
  globalStructure: {
    studentIds: [
      '2tKssy45Qerurvz5T',
      'nWnw9pm9mnrkKNGm3',
      'DAeNWt6uf5y3Kjwqx',
      'nDG9FhEMBsJNt9Mwf',
      'Fff6jCHLLk3aB9oij',
      '6ZLem6dL9CkgLpLBE',
      'DZcvDkx5j3FuG9Zk9',
      'kLDRxiAyQeo5k2rNu'
    ],
    students: {
      '2tKssy45Qerurvz5T': 'Chen Xi',
      nWnw9pm9mnrkKNGm3: 'Per',
      DAeNWt6uf5y3Kjwqx: 'Ahmed',
      nDG9FhEMBsJNt9Mwf: 'Trine',
      Fff6jCHLLk3aB9oij: 'Are',
      '6ZLem6dL9CkgLpLBE': 'Ingunn',
      DZcvDkx5j3FuG9Zk9: 'Kris',
      kLDRxiAyQeo5k2rNu: 'Pierre'
    }
  }
};
