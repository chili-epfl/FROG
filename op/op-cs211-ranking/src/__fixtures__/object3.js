export const config3 = {
  individual: 'cjsth10wk00023f6njzkumn25',
  group: 'cjsth0n6j0007akiy5ummr2qv',
  groupData: 'cjsth0n6j0002akiy13qvhcp2'
};

export const object3 = {
  _id: 'cjsth0n6j000fakiywwe3iimr',
  activityData: {
    cjsth0n6j0007akiy5ummr2qv: {
      structure: { groupingKey: 'group' },
      payload: {
        '1': {
          data: {
            justification: '',
            answers: {
              Dt39sCkvuZC7Ew8iB: {
                Form: 1,
                'Drag and Drop': 2,
                Command: 3,
                Map: 4
              },
              fhCWLQBNJzt8LeQqX: {
                Command: 1,
                'Drag and Drop': 2,
                Form: 3,
                Map: 4
              }
            },
            group: {
              Dt39sCkvuZC7Ew8iB: 'Chen Li',
              fhCWLQBNJzt8LeQqX: 'Aliya'
            },
            coordinates: { x: -1.6, y: 3.2 },
            msg:
              'Uh Oh !! The student assigned to work with you has not completed the previous activity. If he still does not participate, you could discuss with your fellow student sitting next to you.'
          }
        },
        '2': {
          data: {
            justification: '',
            answers: {
              mxJdzWBAfLWJxGF8e: {
                Form: 1,
                'Drag and Drop': 2,
                Command: 3,
                Map: 4
              },
              '8SRWeEbbSwKkZYy8f': {
                Map: 1,
                Command: 2,
                'Drag and Drop': 3,
                Form: 4
              }
            },
            group: {
              mxJdzWBAfLWJxGF8e: 'Peter',
              '8SRWeEbbSwKkZYy8f': 'Anna'
            },
            coordinates: { x: 6.4, y: 3.2 },
            msg:
              'Uh Oh !! The student assigned to work with you has not completed the previous activity. If he still does not participate, you could discuss with your fellow student sitting next to you.'
          }
        }
      }
    },
    cjsth0n6j0002akiy13qvhcp2: {
      structure: { groupingKey: 'group' },
      payload: {
        '1': {
          data: {
            justification: '',
            answers: {
              Dt39sCkvuZC7Ew8iB: {
                Form: 1,
                'Drag and Drop': 2,
                Command: 3,
                Map: 4
              },
              fhCWLQBNJzt8LeQqX: {
                'Drag and Drop': 1,
                Form: 2,
                Command: 3,
                Map: 4
              }
            },
            group: {
              Dt39sCkvuZC7Ew8iB: 'Chen Li',
              fhCWLQBNJzt8LeQqX: 'Aliya'
            },
            coordinates: { x: 6.4, y: -1.6 },
            msg:
              'Uh Oh !! The student assigned to work with you has not completed the previous activity. If he still does not participate, you could discuss with your fellow student sitting next to you.'
          }
        },
        '2': {
          data: {
            justification: '',
            answers: {
              mxJdzWBAfLWJxGF8e: {
                Map: 1,
                Command: 2,
                'Drag and Drop': 3,
                Form: 4
              },
              '8SRWeEbbSwKkZYy8f': { Command: 1 }
            },
            group: {
              mxJdzWBAfLWJxGF8e: 'Peter',
              '8SRWeEbbSwKkZYy8f': 'Anna'
            },
            coordinates: { x: -8, y: 8 },
            msg:
              'Uh Oh !! The student assigned to work with you has not completed the previous activity. If he still does not participate, you could discuss with your fellow student sitting next to you.'
          }
        }
      }
    },
    cjsth10wk00023f6njzkumn25: {
      structure: 'individual',
      payload: {
        '8SRWeEbbSwKkZYy8f': {
          data: {
            answers: {},
            group: {},
            round: 0,
            msg:
              'Uh Oh !! The student assigned to work with you has not completed the previous activity. If he still does not participate, you could discuss with your fellow student sitting next to you.'
          }
        },
        Dt39sCkvuZC7Ew8iB: {
          data: {
            answers: {
              Dt39sCkvuZC7Ew8iB: {
                Form: 1,
                'Drag and Drop': 2,
                'Command Line': 3,
                Map: 4
              }
            },
            group: { Dt39sCkvuZC7Ew8iB: 'Chen Li' },
            round: 1,
            coordinates: { x: 6.4, y: 3.2 },
            alert: false,
            msg:
              'Chen Li ranked the interfaces in the following order: Form, Drag and Drop, Command Line, Map'
          }
        },
        fhCWLQBNJzt8LeQqX: {
          data: {
            answers: {
              fhCWLQBNJzt8LeQqX: {
                Form: 1,
                'Drag and Drop': 2,
                Map: 3,
                'Command Line': 4
              }
            },
            group: { fhCWLQBNJzt8LeQqX: 'Aliya' },
            round: 1,
            coordinates: { x: 6.4, y: 1.6 },
            alert: false,
            msg:
              'Aliya ranked the interfaces in the following order: Form, Drag and Drop, Map, Command Line'
          }
        },
        mxJdzWBAfLWJxGF8e: {
          data: {
            answers: {},
            group: {},
            round: 0,
            msg:
              'Uh Oh !! The student assigned to work with you has not completed the previous activity. If he still does not participate, you could discuss with your fellow student sitting next to you.'
          }
        }
      }
    }
  },
  socialStructure: {},
  globalStructure: {
    studentIds: [
      'Dt39sCkvuZC7Ew8iB',
      'fhCWLQBNJzt8LeQqX',
      'mxJdzWBAfLWJxGF8e',
      '8SRWeEbbSwKkZYy8f'
    ],
    students: {
      Dt39sCkvuZC7Ew8iB: 'Chen Li',
      fhCWLQBNJzt8LeQqX: 'Aliya',
      mxJdzWBAfLWJxGF8e: 'Peter',
      '8SRWeEbbSwKkZYy8f': 'Anna'
    }
  }
};
