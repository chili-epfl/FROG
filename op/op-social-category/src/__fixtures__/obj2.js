export default {
  _id: 'cjvm7pkeq000jl0izs808bxv8',
  activityData: {
    structure: 'individual',
    payload: {
      H52zkjGQ24uruxnST: {
        data: {
          cjvm7ra5r0018l0iz1fhuqry2: {
            id: 'cjvm7ra5r0018l0iz1fhuqry2',
            li: 'cjvm7qk460017l0iz63iz8ghg'
          }
        }
      },
      JKv43553ALfEJBbSw: {
        data: {
          cjvm7ra5r0019l0izp3ehqp1t: {
            id: 'cjvm7ra5r0019l0izp3ehqp1t',
            li: 'cjvm7qk3v0016l0izb6kqi2eg'
          }
        }
      },
      RLoZe2jPGNLqvhWwG: {
        data: {
          cjvm7ra5r001al0iz01jousfs: {
            id: 'cjvm7ra5r001al0iz01jousfs',
            li: 'cjvm7qk3v0015l0izdypbu02r'
          }
        }
      },
      Xz4WL2FEZpApP9aZF: {
        data: {
          cjvm7ra5r001bl0izdsucd477: {
            id: 'cjvm7ra5r001bl0izdsucd477',
            li: 'cjvm7qk3r0011l0izmzigrnbo'
          }
        }
      },
      YhxyDSkKJJvaDnb8K: {
        data: {
          cjvm7ra5r001cl0iz7wyccuua: {
            id: 'cjvm7ra5r001cl0iz7wyccuua',
            li: 'cjvm7qk3s0012l0izbpw5vk1e'
          }
        }
      },
      ZyhkhWTNtFzxLacCx: {
        data: {
          cjvm7ra5r001dl0izlij1jpz4: {
            id: 'cjvm7ra5r001dl0izlij1jpz4',
            li: 'cjvm7qk3p000zl0iz9jotmdsx'
          }
        }
      },
      wd47b9AFR8YYh5tga: {
        data: {
          cjvm7ra5r001el0iztekbvuzj: {
            id: 'cjvm7ra5r001el0iztekbvuzj',
            li: 'cjvm7qk3u0014l0iz4qw18xqn'
          }
        }
      },
      wyjpHTcnfejBfZ6yb: {
        data: {
          cjvm7ra5r001fl0izz7ont0vt: {
            id: 'cjvm7ra5r001fl0izz7ont0vt',
            li: 'cjvm7qk3t0013l0izv3sxdotq'
          }
        }
      },
      xR3CbSik4pS83BWuX: {
        data: {
          cjvm7ra5r001gl0izt9drc16p: {
            id: 'cjvm7ra5r001gl0izt9drc16p',
            li: 'cjvm7qk3q0010l0iz9qa2o373'
          }
        }
      }
    }
  },
  socialStructure: {},
  globalStructure: {
    studentIds: [
      'ZyhkhWTNtFzxLacCx',
      'xR3CbSik4pS83BWuX',
      'Xz4WL2FEZpApP9aZF',
      'YhxyDSkKJJvaDnb8K',
      'wyjpHTcnfejBfZ6yb',
      'wd47b9AFR8YYh5tga',
      'RLoZe2jPGNLqvhWwG',
      'JKv43553ALfEJBbSw'
    ],
    students: {
      ZyhkhWTNtFzxLacCx: 'Chen Li',
      xR3CbSik4pS83BWuX: 'Peter',
      Xz4WL2FEZpApP9aZF: 'Aliya',
      YhxyDSkKJJvaDnb8K: 'Anna',
      wyjpHTcnfejBfZ6yb: 'Alisa',
      wd47b9AFR8YYh5tga: 'Niels',
      RLoZe2jPGNLqvhWwG: 'Natasha',
      JKv43553ALfEJBbSw: 'Bob'
    }
  }
};
