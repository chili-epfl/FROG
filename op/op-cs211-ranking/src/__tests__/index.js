import { object } from '../__fixtures__/object';
import { object2 } from '../__fixtures__/object2';
import pkg from '../index';

const config = {
  individual: 'cjgdgp46h001kblj4cwsqtmrk',
  group: 'cjgdgp46i001mblj4u5lfdd4r',
  groupData: 'cjgdgp46h001lblj4whyzno4p'
};

const text = `Result:

Not completing second activity: 0
Not completing third activity: 1

Transition matrix first->second activity:
{
  "notCompleted": 2,
  "train": {"train": 1, "jump": 1},
  "drag": {"jump": 1},
  "jump": {"train": 1}
}


Transition matrix second->third activity:
{"notCompleted": 2, "train": {"drag": 2}}
`;

test('works', () => {
  expect(pkg.operator(config, object)).toEqual({
    payload: { all: { config: { text }, data: {} } },
    structure: 'all'
  });
});

const text2 = `Result:

Not completing second activity: 2
Not completing third activity: 2

Transition matrix first->second activity:
{
  "notCompleted": 2,
  "plane": {"train": 2},
  "train": {"swimming": 1},
  "swimming": {"swimming": 1}
}


Transition matrix second->third activity:
{"notCompleted": 0, "train": {"train": 2}, "swimming": {"swimming": 2}}
`;

const config2 = {
  individual: 'cjgdu4sti00021sj4sy5xdedy',
  group: 'cjgdu4sti00041sj4v5qch1gk',
  groupData: 'cjgdu4sti00031sj47sjionh9'
};

test('works twice', () => {
  expect(pkg.operator(config2, object2)).toEqual({
    payload: { all: { config: { text: text2 }, data: {} } },
    structure: 'all'
  });
});
