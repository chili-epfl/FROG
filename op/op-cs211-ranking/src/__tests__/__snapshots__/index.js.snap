// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
Object {
  "payload": Object {
    "all": Object {
      "config": Object {
        "text": "
    <div class=\\"op-cs211-ranking\\">
      <div>
        <div class=\\"table-container\\">
          <table>
            <caption> Individual to Group </caption>
            <thead>
              <tr>
                <th></th>
                <th>train</th> <th>drag</th> <th>wave</th> <th>jump</th>
              </tr>
            </thead>
            <tbody>
              <tr key=train>
        <td class=\\"vertical-th\\">train</td>
        <td>1</td> <td>0</td> <td>0</td> <td>1</td>
        </tr> <tr key=drag>
        <td class=\\"vertical-th\\">drag</td>
        <td>0</td> <td>0</td> <td>0</td> <td>1</td>
        </tr> <tr key=wave>
        <td class=\\"vertical-th\\">wave</td>
        <td>0</td> <td>0</td> <td>0</td> <td>0</td>
        </tr> <tr key=jump>
        <td class=\\"vertical-th\\">jump</td>
        <td>1</td> <td>0</td> <td>0</td> <td>0</td>
        </tr>
            </tbody>
          </table>
       </div>
       <div class=\\"table-container\\">
          <table>
             <caption> Group to Group with data </caption>
            <thead>
             <tr>
               <th></th>
              <th>train</th> <th>drag</th> <th>wave</th> <th>jump</th>
             </tr>
            </thead>
            <tbody>
               <tr key=train>
        <td class=\\"vertical-th\\">train</td>
        <td>0</td> <td>2</td> <td>0</td> <td>0</td>
        </tr> <tr key=drag>
        <td class=\\"vertical-th\\">drag</td>
        <td>0</td> <td>0</td> <td>0</td> <td>0</td>
        </tr> <tr key=wave>
        <td class=\\"vertical-th\\">wave</td>
        <td>0</td> <td>0</td> <td>0</td> <td>0</td>
        </tr> <tr key=jump>
        <td class=\\"vertical-th\\">jump</td>
        <td>0</td> <td>0</td> <td>0</td> <td>0</td>
        </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div>
        <p>0 group(s) did not complete the second activity.</p>
        <p>1 group(s) did not complete the third activity.</p>
      </div>
    </div>
    ",
      },
      "data": Object {},
    },
  },
  "structure": "all",
}
`;

exports[`works twice 1`] = `
Object {
  "payload": Object {
    "all": Object {
      "config": Object {
        "text": "
    <div class=\\"op-cs211-ranking\\">
      <div>
        <div class=\\"table-container\\">
          <table>
            <caption> Individual to Group </caption>
            <thead>
              <tr>
                <th></th>
                <th>plane</th> <th>automotive</th> <th>swimming</th> <th>train</th>
              </tr>
            </thead>
            <tbody>
              <tr key=plane>
        <td class=\\"vertical-th\\">plane</td>
        <td>0</td> <td>0</td> <td>0</td> <td>2</td>
        </tr> <tr key=automotive>
        <td class=\\"vertical-th\\">automotive</td>
        <td>0</td> <td>0</td> <td>0</td> <td>0</td>
        </tr> <tr key=swimming>
        <td class=\\"vertical-th\\">swimming</td>
        <td>0</td> <td>0</td> <td>1</td> <td>0</td>
        </tr> <tr key=train>
        <td class=\\"vertical-th\\">train</td>
        <td>0</td> <td>0</td> <td>1</td> <td>0</td>
        </tr>
            </tbody>
          </table>
       </div>
       <div class=\\"table-container\\">
          <table>
             <caption> Group to Group with data </caption>
            <thead>
             <tr>
               <th></th>
              <th>train</th> <th>plane</th> <th>automotive</th> <th>swimming</th>
             </tr>
            </thead>
            <tbody>
               <tr key=train>
        <td class=\\"vertical-th\\">train</td>
        <td>2</td> <td>0</td> <td>0</td> <td>0</td>
        </tr> <tr key=plane>
        <td class=\\"vertical-th\\">plane</td>
        <td>0</td> <td>0</td> <td>0</td> <td>0</td>
        </tr> <tr key=automotive>
        <td class=\\"vertical-th\\">automotive</td>
        <td>0</td> <td>0</td> <td>0</td> <td>0</td>
        </tr> <tr key=swimming>
        <td class=\\"vertical-th\\">swimming</td>
        <td>0</td> <td>0</td> <td>0</td> <td>2</td>
        </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div>
        <p>2 group(s) did not complete the second activity.</p>
        <p>2 group(s) did not complete the third activity.</p>
      </div>
    </div>
    ",
      },
      "data": Object {},
    },
  },
  "structure": "all",
}
`;
